### DISCUSSION TITLE
Infinity to use existing datasources (eg elasticsearch) as backend

### DISCUSSION THREAD

**Discussion by snaeqe** [grafana/grafana-infinity-datasource#259] [Ideas]
Hello,

are there any plans to let Infinity use existing datasources from within Grafana (instead of hardcoding access details into each panel), especially things that do already deliver "compatible" data, like elasticsearch?

We have the Inifinity plugin installed in our Grafana for our data guys, and currently they're using it as a "better elasticsearch frontend", cut&pasting the endpoints of our elasticsearch indices into each Infinity panel config. This of course doesn't adapt to the actual elasticsearch cluster for the respective environment the dashboard is deployed to, breaking it for all environments but the one used to create it..

If there now was the possibility to use the (environment specific) access details already deployed as Grafana datasource, the panel&dashboard would be useable across all environments with identically named datasources, instead of creating one "flavor" of the respective dashboards for each environment and have all those unuseable dashboards show up everywhere.. *rolling my eyes*

Would be glad to hear your opinion on this :) 

best regards,
Sebastian

**Comment by yesoreyeram** [comment]
Hey, thanks for the brilliant suggestions and your experience. Definitely, these kind of feedbacks and inputs help me to understand the usecases and design new features.

I have explored around this long back and found some bottlenecks with the approach. First of all the plugins doesn't have access to credentials stored in other datasources for security reasons. So can't call the endpoints directly. But other workaround is that Plugins can access the query endpoint of other plugins to proxy the requests. With that approach, different plugins handle the data differently. For example some plugins query endpoints send the baked response while others manipulate them at frontend.  So I was not happy with taking both the approaches.

I have similar use cases as you previously but not with elastic search . I had solved that with variables. Different environment have different values in url, query  string and body. https://grafana-infinity-datasource.herokuapp.com/d/infinity-variables/variables-new?orgId=1 is an example of that.

Wondering if any other approach can solve these environments problem.
